arrows/mvg/algo/initialize_cameras_landmarks.cxx

```diff
@@ -2437,7 +2437,8 @@ ::metadata_centric_keyframe_initialization(
       vector_3d pos_loc;
       rotation_d R_loc;
 
-      if (constraints->get_camera_position_prior_local(fid, pos_loc) &&
+      if (constraints == nullptr &&
```

Comment:
This is necessary because the `initialize()` function sets its fourth parameter `constraints` to have a default value of `nullptr`, which as far as I can tell will _always_ result in a segmentation fault in this function . This addition (checking for `nullptr`) stops the segmentation fault, but I believe might cause issues later on with variables not being defined when they should be.

Unless I'm mistaken, nowhere currently calls the `initialize()` function _using_ the default value for `constraints` (doing so would cause a seg fault, I believe), so it would seem the better option here would be to simply change the function to not have a default value for that parameter. This would also necessitate passing in real values for it in the test file

Reply:
Documented this issue in #1639